### DISCUSSION TITLE
formula task count

### DISCUSSION THREAD

**Discussion by DannyHatcher-Git** [RafaelGB/obsidian-db-folder#638] [Q&A] [answered]
Is there a way to get a total for tasks todo/completed for a file?

I know the rollup allows it, and I can have a list of the tasks but I just want the number.

My guess is a formula could do it but I am not sure how.

**Comment by RafaelGB** [comment, marked as answer]
There is not a direct way to obtain that number, but you can use dataview


```javascript
${db.dataview.page(row.__file__.path).file.tasks.length}
```

**Reply by DannyHatcher-Git** [reply]
ooh ok. I didn't know we could do that in the formula box.

The formula property seem extreamely powerful but very confusing 😆

**Reply by RafaelGB** [reply]
yeah, its designed to run javascript. You can really do whatever you want inside. Over time we will add syntactic sugar to make it more friendly

**Reply by DannyHatcher-Git** [reply]
is it a case of learning js for using the formula box right now?

**Reply by RafaelGB** [reply]
with very basic cases it would not be necessary but if you want to start doing complex things you have the option with js

**Comment by DannyHatcher-Git** [comment]
if i want it to measure todo tasks, how would i edit it?

**Reply by RafaelGB** [reply]
You can check the dataview API info. In your case:
```javascript
// ...
file.tasks.where(t => !t.completed).length
```
